### PR TITLE
Validate silent-Playbook and silent-trigger relationships

### PR DESCRIPTION
**Pull request by moishce**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-12154

## Description
A silent-trigger must point to a silent-Playbook, and a silent-Playbook must point to a silent-trigger.

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/PB_validators/PB131_is_silent_playbook_relationships.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+
+class IsSilentPlaybookRelationshipsValidator(BaseValidator[ContentTypes]):
+    error_code = "PB131"
+    description = "A silent-Playbook/Trigger must point to a silent-Playbook/Trigger"
+    rationale = "A silent-Playbook/Trigger must point to a silent-Playbook/Trigger"
+    error_message = (
+        "Your silent-Playbook/Trigger does not point to a silent-Playbook/Trigger"
+    )
+    related_field = ""
```
Comment (dantavori) [inline demisto_sdk/commands/validate/validators/PB_validators/PB131_is_silent_playbook_relationships.py:23]:
```suggestion
    description = "Every silent trigger must correspond to a silent playbook in the same pack, and vice versa."
    rationale = "To ensure the effective operation of the silent playbook items."
    error_message = (
        "The {} is silent, but does not correspond to a silent {} in the pack."
    )
    related_field = "isSilent"
```
note my addition of "in the pack" as this is what you check in the code. this is not exactly the definition in the ticket - I guess it's good enough but let's ensure that.

File: demisto_sdk/commands/validate/validators/PB_validators/PB131_is_silent_playbook_relationships.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    is_auto_fixable = False
+
+    def obtain_invalid_content_items(
+        self, content_items: Iterable[ContentTypes]
+    ) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message,
```
Comment (dantavori) [inline demisto_sdk/commands/validate/validators/PB_validators/PB131_is_silent_playbook_relationships.py:32]:
```suggestion
    ) -> List[ValidationResult]:
        def get_types_for_err_msg(c: ContentItem) -> tuple[ContentTypes, ContentTypes]:
            if isinstance(c, Trigger):
                return ContentTypes.TRIGGER, ContentTypes.PLAYBOOK
            return ContentTypes.PLAYBOOK, ContentTypes.TRIGGER
        return [
            ValidationResult(
                validator=self,
                message=self.error_message.format(*get_types_for_err_msg(content_item)),
```
just to make the error message a bit nicer - please verify the change

File: demisto_sdk/commands/validate/tests/PB_validators_test.py [hunk 3]
```diff
@@ -1423,3 +1430,155 @@ def test_IsSilentPlaybookValidator(name, id, is_silent, result_len):
...
+    Given:
+    - case 1: faund the playbook that the trigger points on, but the playbook is not silent.
+    - case 2: faund the playbook that the trigger points on, and both are silent.
+    - case 3: not faund the playbook that the trigger points on.
+    - case 2: not faund the playbook that the trigger points on.
+    - case 1: but the trigger is not silent.
+    When:
+    - Calling IsSilentPlaybookRelationshipsValidator for playbooks.
+    Then:
+    - Validate that only invalid items are returned.
```
Comment (dantavori) [inline demisto_sdk/commands/validate/tests/PB_validators_test.py:1567]:
```suggestion
    Given:
    - case 1: A silent trigger that points on a non-silent playbook.
    - case 2: A silent trigger that points on a silent playbook.
    - case 3: A silent trigger that points on a non-silent playbook that is not found.
    - case 4: A silent trigger that points on a silent playbook that is not found.
    - case 5: A non-silent trigger that points on a silent playbook.
    When:
    - Calling IsSilentPlaybookRelationshipsValidator for playbooks.
    Then:
    - Validate that only invalid items are returned.
```

File: demisto_sdk/commands/validate/tests/PB_validators_test.py [hunk 4]
```diff
@@ -1423,3 +1430,155 @@ def test_IsSilentPlaybookValidator(name, id, is_silent, result_len):
...
+def test_IsSilentPlaybookRelationshipsValidator(
+    playbook_id, playbook_is_silent, trigger_playbook_id, trigger_is_silent, result_len
+):
+    """
+    Given:
+    - case 1: faund a trigger that points on the playbook but the trigger is not silent.
+    - case 2: faund a trigger that points on the playbook and both are silent.
+    - case 3: not faund a trigger that points on the playbook.
+    - case 2: not faund a trigger that points on the playbook.
+    - case 1: the playbook is not silent.
```
Comment (dantavori) [inline demisto_sdk/commands/validate/tests/PB_validators_test.py:1488]:
```suggestion
    - case 1: A silent playbook that corresponds to a non-silent trigger.
    - case 2: A silent playbook that corresponds to a silent trigger.
    - case 3: A silent playbook that corresponds to a non-silent trigger that is not found.
    - case 4: A silent playbook that corresponds to a silent trigger that is not found.
    - case 5: A non-silent playbook that corresponds to a silent trigger.
```

**Review by dantavori — COMMENTED** [review]
nice, see minor notes

File: demisto_sdk/commands/validate/validators/PB_validators/PB131_is_silent_playbook_relationships.py (hunk 1 above)
Comment (moishce) [reply demisto_sdk/commands/validate/validators/PB_validators/PB131_is_silent_playbook_relationships.py]: Yes, I verified that, and that's what's needed.

**Review by dantavori — APPROVED** [review]
nice

File: demisto_sdk/commands/validate/validators/PB_validators/PB131_is_silent_playbook_relationships.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            for content_item in content_items
+            if not is_valid_relationship(content_item)
+        ]
+
+
+def is_valid_relationship(content_item: ContentTypes) -> bool:
+    """
+    Validates the relationship between a content item and its associated playbook/trigger.
+    """
+    if not content_item.data.get("isSilent", False):
```
Comment (dantavori) [inline demisto_sdk/commands/validate/validators/PB_validators/PB131_is_silent_playbook_relationships.py:50]: although data is cached, wouldn't it be better to just use is_silent property directly?

**Pull request merged**